### PR TITLE
Add gzip and server_names_hash_bucket_size configuration

### PR DESCRIPTION
**Pull request by itskingori**
See commits for details. Hard to see changes clearly due to the diff but ... follow commits for an easier review as each commit is an idea.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 1.19.5-1
+
+* Add `ngx_http_gzip_module` configuration.
+* Add `server_names_hash_bucket_size` configuration.
```
Comment (zacblazic) [inline CHANGELOG.md:6]: I left out `server_names_hash_bucket_size` as it's only necessary to tweak if we have a large number of upstreams, which is not likely to be the case when this is going to be a sidecar.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 1.19.5-1
+
+* Add `ngx_http_gzip_module` configuration.
```
Comment (zacblazic) [inline CHANGELOG.md:5]: I left out compression as I assumed that we'd let the application or load balancer handle it. I'm not against adding this if neither are able to be configured with compression.

File: CHANGELOG.md (hunk 1 above)
Comment (itskingori) [reply CHANGELOG.md:6]: I see. Fair enough.

File: CHANGELOG.md (hunk 2 above)
Comment (itskingori) [reply CHANGELOG.md:5]: Right. I'm ok with that.

**Comment by itskingori** [comment]
Closing as there were good reasons why these configurations were excluded.

**Pull request closed without merging**